Add docs and licenses (#1)
DataDog/dd-sdk-ios-apollo-interceptor: README.md

# Datadog Integration for Apollo iOS

## Getting started

**Note:** The integration supports Apollo iOS version 1.0+.

To include the integration for [Apollo iOS](https://github.com/apollographql/apollo-ios) in your project, add the following to your `Package.swift` file:

```swift
dependencies: [
    .package(url: "https://github.com/DataDog/dd-sdk-ios-apollo", .upToNextMajor(from: "1.0.0"))
]
```

> Or add it through Xcode:

Reviewer: ```suggestion
Alternatively, you can add it using Xcode:
```

1. Go to File → Add Package Dependencies
2. Enter the repository URL: `https://github.com/DataDog/dd-sdk-ios-apollo`
3. Select the latest version

## Initial setup

1. Set up RUM monitoring with [Datadog iOS RUM SDK](https://docs.datadoghq.com/real_user_monitoring/ios/)

> 2. Set up network instrumentation with the Datadog RUM SDK for iOS

Reviewer 2: question: Why does the apollo support logic needs a different repo if it is dependent on the iOS RUM SDK (meaning it won't work in isolation)?

Author: Good question! The main reason is dependency management. The Apollo package itself is quite heavy and if we included the integration as a module inside the iOS SDK, the Swift Package Manager would force-install Apollo for everyone, even customers who don't necessarily use it.
By keeping the integration in a separate repository, we ensure that only customers who actually use Apollo need to pull in this dependency.

Reviewer 2: If the module that contains the Apollo sdk is not linked into the app, it won't be installed. 
Being in the project side bar does not mean it is installed and linked in an app (we have the Otel and PLCrashReporter as good examples). 
There is a good new feature in SPM ([packagetraits](https://docs.swift.org/swiftpm/documentation/packagemanagerdocs/packagetraits/)), introduced along with Swift 6.1, that removes this optional dependencies from the side bar. 
I did a quick test, and it works fine if the project app controls the injection of the trait when importing the `dd-sdk-ios`.

Reviewer 3: It is the same problem with Otel: even if the package is not linked, SPM resolves all transitive dependencies. We don't want that.

> I did a quick test, and it works fine if the project app controls the injection of the trait when importing

How does the app can control de trait in Xcode?

Reviewer 2: The app project can control it by including its own Package. For now, the UI does not have an option to add traits when adding a package.

3. Add the Datadog interceptor to your Apollo Client setup:

```swift
import Apollo
import DatadogApollo

class CustomInterceptorProvider: DefaultInterceptorProvider {
    override func interceptors<Operation: GraphQLOperation>(for operation: Operation) -> [ApolloInterceptor] {
        var interceptors = super.interceptors(for: operation)
        interceptors.insert(DatadogApollo.createInterceptor(), at: 0)
        return interceptors
    }
}
```

> This automatically adds Datadog headers to your GraphQL requests, allowing them to be tracked by Datadog. Note that while `query` and `mutation` type operations are tracked, `subscription` operations are not.

Reviewer: ```suggestion
This automatically adds Datadog headers to your GraphQL requests, enabling them to be tracked by Datadog. Note that while query and mutation operations are tracked, subscription operations are not.
```

## Sending GraphQL payloads

> GraphQL payload sending is disabled by default. To enable it, set the `sendGraphQLPayloads` flag in the DatadogApollo interceptor constructor as follows:

Reviewer: ```suggestion
Sending GraphQL payloads is disabled by default. To enable it, set the `sendGraphQLPayloads` flag in the DatadogApollo interceptor constructor as shown below:
```

```swift
let datadogInterceptor = DatadogApollo.createInterceptor(sendGraphQLPayloads: true)
```

## Contributing

> For details on contributing, read the [Contributing Guide](CONTRIBUTING.md).

Reviewer: ```suggestion
Contributions are welcome! For details, see the [Contributing Guide](CONTRIBUTING.md).
```
